libbeat/autodiscover/autodiscover_test.go

```diff
@@ -570,6 +624,138 @@ func TestAutodiscoverWithMutlipleEntries(t *testing.T) {
 	check(t, runners, conf.MustNewConfigFrom(map[string]interface{}{"x": "c"}), false, true)
 }
 
+func TestAutodiscoverDebounce(t *testing.T) {
+	printDebugLogsOnFailure(t)
+	// Register mock autodiscover provider
+	busChan := make(chan bus.Bus, 1)
+	Registry = NewRegistry()
+	err := Registry.AddProvider("mock", func(beatName string, b bus.Bus, uuid uuid.UUID, c *conf.C, k keystore.Keystore) (Provider, error) {
+		// intercept bus to mock events
+		busChan <- b
+
+		return &mockProvider{}, nil
+	})
+	if err != nil {
+		t.Fatalf("cannot add provider to registry: %s", err)
+	}
+
+	providerConfig, _ := conf.NewConfigFrom(map[string]string{
+		"type": "mock",
+	})
+	config := Config{
+		Providers: []*conf.C{providerConfig},
+	}
+	k, _ := keystore.NewFileKeystore("test")
+
+	adapter := mockAdapter{}
+
+	// Create autodiscover manager
+	autodiscover, err := NewAutodiscover("test", nil, &adapter, &adapter, &config, k)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// Start it
+	autodiscover.Start()
+	t.Cleanup(autodiscover.Stop)
+
+	eventBus := <-busChan
+
+	cfg, err := conf.NewConfigFrom(map[string]string{
+		"foo": "bar",
+	})
+	if err != nil {
+		t.Fatalf("error creating input config: %s", err)
+	}
+
+	// Send an event with config,
+	// `Autodiscover.handleStart` will return true and
+	// updated will be set to true
+	eventBus.Publish(bus.Event{
+		"id":       "foo",
+		"provider": "mock",
+		"start":    true,
+		"meta": mapstr.M{
+			"foo": "bar",
+		},
+		"config": []*conf.C{cfg},
+	})
+
+	// Send the second event without a config
+	// `Autodiscover.handleStart` will return false and
+	// updated must not be changed
+	eventBus.Publish(bus.Event{
+		"id":       "second,",
+		"provider": "mock",
+		"start":    true,
+		"meta": mapstr.M{
+			"foo": "bar",
+		},
+	})
+
+	// Ensure the input has been started
+	requireRunningRunners(t, autodiscover, 1)
+
+	// Repeat the process, but this time with a stop event.
+	// The same logic applies, but now we're testing the branch that calls
+	// `Autodiscover.handleStop`
+	eventBus.Publish(bus.Event{
+		"id":       "foo",
+		"provider": "mock",
+		"stop":     true,
+		"meta": mapstr.M{
+			"foo": "bar",
+		},
+		"config": []*conf.C{cfg},
+	})
+	eventBus.Publish(bus.Event{
+		"id":       "second,",
+		"provider": "mock",
+		"stop":     true,
+		"meta": mapstr.M{
+			"foo": "bar",
+		},
+	})
+	requireRunningRunners(t, autodiscover, 0)
+}
+
+func printDebugLogsOnFailure(t *testing.T) {
```

Comment:
Should this have a `t.Helper()`?

Reply:
I'm on the fence with this one.

It is a helper that does not do any assertion, so if it fails it's most likely not related to the test itself, but to its own implementation. So my first thought is that it would be more helpful to see the line that failed rather than where `printDebugLogsOnFailure` was called.

What do you think?